test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/Common/CustomHttpClientRequestMessage.cs

```diff
@@ -0,0 +1,127 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="CustomHttpClientRequestMessage.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.E2E.TestCommon.Common
+{
+    /// <summary>
+    /// An implementation of IODataRequestMessageAsync that uses an HttpRequestMessage under the covers.
+    /// In ODataLibrary, a message is an abstraction which consists of stream and header interfaces that hides the details of stream-reading/writing.
+    /// </summary>
+    public class CustomHttpClientRequestMessage : IODataRequestMessageAsync, IServiceCollectionProvider

```

Comment:
Should this class perhaps implement `IDisposable` and `IAsyncDisposable` to clean up disposable objects like the stream object?